### PR TITLE
Allow installation of Dapr runtime into a specific Docker network

### PR DESCRIPTION
**Pull request by philliphoff**
# Description

Enables installation of the Dapr runtime into a specific Docker network (e.g. for use in [VS Code Dev Container](https://code.visualstudio.com/docs/remote/containers) scenarios). When doing so, the Dapr service containers are given network-specific container names to allow multiple runtime instances as well as distinguish between runtime instances when uninstalling.  The Dapr service containers are also given well-known network aliases to allow following `dapr run` commands made within that network to specify those, now non-`localhost` hosts

Specific changes:

 - Add `.vscode` to `.gitignore`
   - Allows user/platform specific build/debugging configuration
 - Add optional `--network` argument to `dapr init` command
   - If specified, injects the Redis and placement service containers into the specified Docker network and does **not** map container ports to the local machine
   - If omitted, preserves the existing behavior (i.e. the service containers are started without a specific network and container ports **are** mapped to the local machine)
 - Add optional `--redis-host` and `--placement-host` arguments to the `dapr run` command
   - If specified, uses the arguments as the host names for the Redis and placement service addresses, respectively, passed to the underlying `daprd` service (directly or indirectly via the component YAML files)
   - If omitted, preserves the existing behavior (i.e. both host names default to `localhost`)
 - Add optional `--network` argument to `dapr uninstall` command
   - If specified, removes the Redis and placement service containers from the specified Docker network
   - If omitted, preserves the existing behavior (i.e. the service containers not in any specific network are removed)
 - Updates `README.md` to provide examples of using these new arguments

## Issue reference

This PR resolves issue #191.

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [ ] Created/updated tests
* [x] Extended the documentation

### CODE REVIEW

**Comment by yaron2** [comment]
From first glance everything looks great. I'll take on a full review later on. Thanks @philliphoff

File: utils/utils.go [hunk 1]
```diff
@@ -67,3 +68,11 @@ func RunCmdAndWait(name string, args ...string) error {
 
 	return nil
 }
+
+func CreateContainerName(serviceContainerName string, dockerNetwork string) string {
+	if (dockerNetwork != "") {
+		return fmt.Sprintf("%s_%s", serviceContainerName, dockerNetwork)
+	} else {
+		return serviceContainerName
```
Comment (yaron2) [inline utils/utils.go]: Remove the else block

**Comment by yaron2** [comment]
One comment + resolve conflict and we're good to go.

File: utils/utils.go (hunk 1 above)
Comment (philliphoff) [reply utils/utils.go]: Updated.

**Comment by yaron2** [comment]
Thanks for this @philliphoff!

**Pull request merged**